workers/unity/Packages/com.improbable.gdk.tools/DownloadCoreSdk.cs

```diff
@@ -136,6 +142,42 @@ private static DownloadResult Download()
             return exitCode == 0 ? DownloadResult.Success : DownloadResult.Error;
         }
 
+        private static bool CheckDependencies()
+        {
+            var hasDotnet = Common.DotNetBinary != string.Empty;
+            var hasSpatial = Common.SpatialBinary != string.Empty;
+
+            if (hasDotnet && hasSpatial)
+            {
+                return true;
+            }
+
+            var builder = new StringBuilder();
+
+            if (!hasDotnet)
+            {
+                builder.AppendLine("Could not find dotnet on your PATH.");
+            }
+
+            if (!hasSpatial)
+            {
+                builder.AppendLine("Could not find spatial on your PATH.");
+            }
+
+            builder.AppendLine();
+            builder.AppendLine("If these exist on your path, restart Unity and Unity Hub.");
+            builder.AppendLine();
+            builder.AppendLine("Otherwise, install them by following our setup guide: ");
+            builder.AppendLine("https://docs.improbable.io/unity/alpha/get-started#set-up-your-machine");
+
+            EditorApplication.delayCall += () =>
+            {
+                EditorUtility.DisplayDialog("Dependency check failed", builder.ToString(), "OK");
```

Comment:
"Dependency check failed" is ambiguous, can we be explicit in the title and the first line of the message? Especially if this will run automatically when they open the project.

Reply:
What would you prefer here?